src/controllers/connection/newConnection.ts

```diff
@@ -0,0 +1,75 @@
+import { Get, Produces, Query, Route, SuccessResponse } from 'tsoa'
+import { inject, injectable, singleton } from 'tsyringe'
+
+import { Logger, type ILogger } from '../../logger.js'
+import CompantHouseEntity from '../../models/companyHouseEntity.js'
+import NewConnectionTemplates from '../../views/newConnection.js'
+import { HTML, HTMLController } from '../HTMLController.js'
+
+@singleton()
+@injectable()
+@Route('/new-connection')
+@Produces('text/html')
+export class NewConnectionController extends HTMLController {
+  constructor(
+    private companyHouseEntity: CompantHouseEntity,
+    private newConnection: NewConnectionTemplates,
+    @inject(Logger) private logger: ILogger
+  ) {
+    super()
+    this.logger = logger.child({ controller: '/' })
+  }
+
+  /**
+   *
+   * @returns The new connections form page
+   */
+  @SuccessResponse(200)
+  @Get('/')
+  public async newConnectionForm(): Promise<HTML> {
+    this.logger.debug('new connection page requested')
+    return this.html(this.newConnection.formPage())
+  }
+
+  /**
+   * Returns a company from a validated company number
+   */
+  @SuccessResponse(200)
+  @Get('/verify-company')
+  public async verifyCompanyForm(@Query() companyNumber: string): Promise<HTML> {
```

Comment:
as discussed we should probably make this replace the whole form. It'll just make further validation easier